apps/src/applab/designMode.js

```diff
@@ -369,3 +367,56 @@ function makeDraggable (jq) {
     }
   });
 }
+
+designMode.configureDragAnDrop = function () {
+  // Allow elements to be dragged and dropped from the design mode
+  // element tray to the play space.
+  $('.new-design-element').draggable({
+    containment:"#codeApp",
+    helper:"clone",
+    appendTo:"#codeApp",
+    revert: 'invalid',
+    zIndex: 2,
+    start: function() {
+      studioApp.resetButtonClick();
+    }
+  });
+  var GRID_SIZE = 5;
+  $('#visualization').droppable({
+    accept: '.new-design-element',
+    drop: function (event, ui) {
+      var elementType = ui.draggable[0].dataset.elementType;
+
+      var div = document.getElementById('divApplab');
+      var xScale = div.getBoundingClientRect().width / div.offsetWidth;
+      var yScale = div.getBoundingClientRect().height / div.offsetHeight;
+
+      var left = ui.position.left / xScale;
+      var top = ui.position.top / yScale;
+
+      // snap top-left corner to nearest location in the grid
+      left -= (left + GRID_SIZE / 2) % GRID_SIZE - GRID_SIZE / 2;
+      top -= (top + GRID_SIZE / 2) % GRID_SIZE - GRID_SIZE / 2;
+
+      designMode.createElement(elementType, left, top);
+    }
+  });
+};
+
+designMode.configureDesignToggleRow = function () {
+  var designToggleRow = document.getElementById('designToggleRow');
+  if (!designToggleRow) {
+    return;
+  }
+
+  // TODO (brent) - still need logic to generate list of screens, and rerender
+  // DesignToggleRow on changes
+  React.render(
+    React.createElement(DesignToggleRow, {
```

Comment:
The design button (or perhaps this whole component) needs to be hidden behind user.isAdmin . This is how we are hiding unlaunched features from field testers for now.